modules/local_storage/src/LocalStockUpdater.php

```diff
@@ -58,21 +59,25 @@ public static function create(ContainerInterface $container) {
   /**
    * {@inheritdoc}
    */
-  public function createTransaction(PurchasableEntityInterface $entity, $location_id, $zone, $quantity, $unit_cost, $transaction_type_id, array $metadata) {
+  public function createTransaction(PurchasableEntityInterface $entity, $location_id, $zone, $quantity, Price $unit_cost, $transaction_type_id, array $metadata) {
     // Get optional fields.
     $related_tid = isset($metadata['related_tid']) ? $metadata['related_tid'] : NULL;
     $related_oid = isset($metadata['related_oid']) ? $metadata['related_oid'] : NULL;
     $related_uid = isset($metadata['related_uid']) ? $metadata['related_uid'] : NULL;
     $data = isset($metadata['data']) ? $metadata['data'] : NULL;
 
+    $cost_number = $unit_cost->isZero() ? NULL : $unit_cost->getNumber();
+    $cost_code = $unit_cost->isZero() ? NULL : $unit_cost->getCurrencyCode();
+
```

Comment:
remove this for now -- save price even if amount is zero and currency code is XXX.